public/app/features/dashboard/components/GenAI/GenAIButton.tsx

```diff
@@ -26,38 +28,89 @@ export interface GenAIButtonProps {
 export const GenAIButton = ({
   text = 'Auto-generate',
   loadingText = 'Generating',
+  toggleTipTitle = '',
   onClick: onClickProp,
   messages,
   onGenerate,
   temperature = 1,
 }: GenAIButtonProps) => {
   const styles = useStyles2(getStyles);
 
-  // TODO: Implement error handling (use error object from hook)
-  const { setMessages, reply, isGenerating, value, error } = useOpenAIStream(OPEN_AI_MODEL, temperature);
+  const [history, setHistory] = useState<string[]>([]);
+  const [shouldCloseHistory, setShouldCloseHistory] = useState(false);
+
+  const { setMessages, reply, value, error, streamStatus } = useOpenAIStream(OPEN_AI_MODEL, temperature);
+
+  const hasHistory = history.length > 0;
+  const isFirstGeneration = streamStatus === StreamStatus.GENERATING && !hasHistory;
```

Comment:
```suggestion
  const isFirstHistoryEntry = streamStatus === StreamStatus.GENERATING && !hasHistory;
```
Just to keep the same naming :)